client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java

```diff
@@ -0,0 +1,201 @@
+    threadPoolExecutor = (ThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    maxPoolSize = threadPoolExecutor.getMaximumPoolSize();
+    maxThreadIdleMilliseconds = threadPoolExecutor.getKeepAliveTime(TimeUnit.MILLISECONDS);
+    Assert.assertEquals("Wrong life cycle executor core pool size", DEFAULT_CORE_POOL_SIZE, corePoolSize);
+    Assert.assertEquals("Wrong life cycle executor maximum pool size", DEFAULT_MAX_POOL_SIZE, maxPoolSize);
+    Assert.assertEquals("Wrong life cycle executor maximum pool size", DEFAULT_MAX_THREAD_IDLE_MILLISECONDS, maxThreadIdleMilliseconds);
+
+    executorService = executorContext.getApiExecutor();
+    Assert.assertTrue(executorService instanceof ThreadPoolExecutor);
+    threadPoolExecutor = (ThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    maxPoolSize = threadPoolExecutor.getMaximumPoolSize();
+    maxThreadIdleMilliseconds = threadPoolExecutor.getKeepAliveTime(TimeUnit.MILLISECONDS);
+    Assert.assertEquals("Wrong api executor core pool size", DEFAULT_CORE_POOL_SIZE, corePoolSize);
+    Assert.assertEquals("Wrong api executor maximum pool size", DEFAULT_MAX_POOL_SIZE, maxPoolSize);
+    Assert.assertEquals("Wrong api executor maximum pool size", DEFAULT_MAX_THREAD_IDLE_MILLISECONDS, maxThreadIdleMilliseconds);
+
+    executorService = executorContext.getCallbackExecutor();
+    Assert.assertTrue(executorService instanceof ThreadPoolExecutor);
+    threadPoolExecutor = (ThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    maxPoolSize = threadPoolExecutor.getMaximumPoolSize();
+    maxThreadIdleMilliseconds = threadPoolExecutor.getKeepAliveTime(TimeUnit.MILLISECONDS);
+    Assert.assertEquals("Wrong callback executor core pool size", DEFAULT_CORE_POOL_SIZE, corePoolSize);
+    Assert.assertEquals("Wrong callback executor maximum pool size", DEFAULT_MAX_POOL_SIZE, maxPoolSize);
+    Assert.assertEquals("Wrong callback executor maximum pool size", DEFAULT_MAX_THREAD_IDLE_MILLISECONDS, maxThreadIdleMilliseconds);
+
+    executorService = executorContext.getScheduledExecutor();
+    Assert.assertTrue(executorService instanceof ScheduledThreadPoolExecutor);
+    threadPoolExecutor = (ScheduledThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    Assert.assertEquals("Wrong scheduled executor core pool size", DEFAULT_MAX_POOL_SIZE, corePoolSize);
+  }
+
+  @Test
+  public void testConstructorWithAllParameters() throws Exception {
+    executorContext = new FlexibleExecutorContext(CUSTOM_LIFE_CYCLE_THREAD_COUNT, CUSTOM_API_THREAD_COUNT, CUSTOM_CALLBACK_THREAD_COUNT, CUSTOM_SCHEDULED_THREAD_COUNT);
+    executorContext.init();
+
+    ExecutorService executorService;
+    ThreadPoolExecutor threadPoolExecutor;
+    long corePoolSize, maxPoolSize, maxThreadIdleMilliseconds;
```

Comment:
Use **int** type as in constants.
Declare every variable in separate line. No variable grouping is recommended.